src/main/java/sirius/db/mongo/Mango.java

```diff
@@ -138,6 +142,27 @@ protected void updateEntity(MongoEntity entity, boolean force, EntityDescriptor
         }
     }
 
+    private void writeField(MongoEntity entity, Updater updater, Property property) {
+        Object valueForDatasource = property.getValueForDatasource(Mango.class, entity);
+        if (isDefaultValue(valueForDatasource) && property.isAnnotationPresent(SkipDefaultValue.class)) {
+            updater.unset(property.getPropertyName());
+        } else {
+            updater.set(property.getPropertyName(), valueForDatasource);
+        }
+    }
+
+    /**
+     * Determines if the given value is the default value.
+     * <p>
+     * This is the value which is also assumed if no value at all is present in the database.
+     *
+     * @param valueForDatasource the value to check
+     * @return <tt>true</tt> if the given value is a default value, <tt>false</tt> otherwise
+     */
+    private boolean isDefaultValue(Object valueForDatasource) {
+        return valueForDatasource == null || Boolean.FALSE.equals(valueForDatasource);
```

Comment:
Kann auch falsch liegen, aber ist es nicht so dass man bei Entities einem Feld auch nen wirklichen default Wert geben kann, indem man dem Feld direkt bei der Definition einen Wert mitgibt?

Also sowas in die Richtung meine ich:
```
public static final Mapping HAS_OTHER_DEFAULT_VALUE = Mapping.named("hasOtherDefaultValue");
private String hasOtherDefaultValue = "blub";
```

Mit der aktuellen Logik des PRs würde das ja ignoriert und einfach gespeichert oder? Fände ich okay, sollte man aber evtl im JavaDoc erwähnen oder?

Reply:
Yep, this would be saved in the DB. "Default Value" in this case only matches null and false which are the default within the Property implementation - so for both, the given value and a missing value, the property will derive the same value and write it back into the entity.